github/actions/client.go

```diff
@@ -165,6 +165,29 @@ func NewClient(githubConfigURL string, creds *ActionsAuth, options ...ClientOpti
 	return ac, nil
 }
 
+// Identifier returns a string to help identify a client uniquely.
+// This is used for caching client instances and understanding when a config
+// change warrants creating a new client. Any changes to Client that would
+// require a new client should be reflected here.
+func (c *Client) Identifier() string {
+	identifier := fmt.Sprintf("configURL:%s,", c.config.ConfigURL.String())
```

Comment:
Suggestion (ignore me if you don't agree :relaxed:)
Should we use %q  for strings? It would not add any value for key since it is being hashed but it might be easier for debugging

Reply:
Actually very nice, thank you!